rest/src/main/java/org/jboss/pnc/rest/endpoint/ProductMilestoneEndpoint.java

```diff
@@ -203,6 +203,30 @@ public Response update(
         return Response.ok().build();
     }
 
+    @ApiOperation(value = "Close/Release a Product Milestone")
+    @ApiResponses(value = {
+            @ApiResponse(code = SUCCESS_CODE, message = SUCCESS_DESCRIPTION),
+            @ApiResponse(code = INVALID_CODE, message = INVALID_DESCRIPTION, response = ErrorResponseRest.class),
+            @ApiResponse(code = CONFLICTED_CODE, message = CONFLICTED_DESCRIPTION, response = ErrorResponseRest.class),
+            @ApiResponse(code = SERVER_ERROR_CODE, message = SERVER_ERROR_DESCRIPTION, response = ErrorResponseRest.class)
+    })
+    @PUT
+    @Path("/{id}/close-milestone")
```

Comment:
What endpoint was used before to close the milestone and start the milestone release process?

Reply:
Before we used the `update` endpoint. We had to set the end date to start the milestone release process.

Since we now set the end date to the date when the release is done, I thought it'd be better to create a specific endpoint to just close the milestone.

It'll check that:
- the end date is not already set (which would mean the release is already done)
- there are no release in progress (there would be a release in progress if we hit the 'Close Milestone' button 2 times in quick succession). If there is a release in progress, do nothing.
- If there are no release in progress, trigger the release
- In the release process code, if the process is successful, it'll set the end date

Reply:
Great! That's a good way of doing that.